proto/desmos/supply/v1/query.proto

```diff
@@ -0,0 +1,73 @@
+syntax = "proto3";
+package desmos.supply.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "google/protobuf/any.proto";
+import "cosmos_proto/cosmos.proto";
+import "cosmos/base/v1beta1/coin.proto";
+
+option go_package = "github.com/desmos-labs/desmos/v3/x/supply/types";
+
+
+// Query defines the gRPC querier service.
+service Query {
+
+  // TotalSupply queries the total supply of the given denom, converted with the given divider
+  rpc TotalSupply(QueryTotalSupplyRequest)
+      returns (QueryTotalSupplyResponse) {
+    option (google.api.http).get = "/desmos/supply/v1/total-supply/{denom}";
+  }
+
+  // CirculatingSupply queries the amount of tokens circulating in the market of the given
+  rpc CirculatingSupply(QueryCirculatingSupplyRequest)
+      returns (QueryCirculatingSupplyResponse) {
+    option (google.api.http).get = "/desmos/supply/v1/circulating-supply/{denom}";
+  }
+}
+
+// QueryTotalSupplyRequest is the request type for Query/TotalSupply RPC method
+message QueryTotalSupplyRequest {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // coin denom to query the circulating supply for
+  string denom = 1;
+
+  // divider_exponent is a factor used to power the divider used to convert the supply to the desired representation
+  uint64 divider_exponent = 2;
+}
+
+// QueryTotalSupplyResponse is the response type for the Query/TotalSupply RPC method
+message QueryTotalSupplyResponse {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  string total_supply = 1 [
+    (gogoproto.customtype) = "github.com/cosmos/cosmos-sdk/types.Int",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// QueryCirculatingSupplyRequest is the request type for the Query/CirculatingSupply RPC method
+message QueryCirculatingSupplyRequest {
+  option (gogoproto.equal) = false;
+  option (gogoproto.goproto_getters) = false;
+
+  // coin denom to query the circulating supply for
+  string denom = 1;
+
+  // divider_exponent is a factor used to power the divider used to convert the supply to the desired representation
+  uint64 divider_exponent = 2;
```

Comment:
It could be uint8.

Reply:
Protobuf 3 only supports `uint32` or `uint64`. I think there's no much difference if we leave it as it is